src/hotspot/share/code/nmethod.cpp

```diff
@@ -1387,10 +1388,285 @@ nmethod::nmethod(
   }
 }
 
+
+nmethod::nmethod(nmethod* nm) : CodeBlob(nm->_name, nm->_kind, nm->_size, nm->_header_size)
```

Comment:
Should this be a copy constructor? 
```
nmethod::nmethod(const nmethod &nm) : CodeBlob(nm._name, nm._kind, nm._size, nm._header_size)
```
Even if we don't make it a copy constructor, it looks like its nmethod argument should be `const`, but I haven't checked very deeply.